Move interned_strings and get build working

bwasti opened the pull request:
Summary: Refactoring out this diff D9819906

Differential Revision: D10024844

ezyang commented:
Please ping when you want review

Yangqing commented on aten/src/ATen/core/interned_strings_class.h:
```diff
@@ -0,0 +1,39 @@
+namespace torch {
+namespace jit {
+
+struct CAFFE2_API InternedStrings {
```
Sorry, I am going to be a documentation policeman for a while but this class has zero comments :)

bwasti replied:
that's true https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/interned_strings_class.h

ezyang replied:
```
// This class represents the global state that keeps track of interned strings.
// It maintains mappings of strings to their interned representation (Symbol)
// and vice versa.
```

ezyang commented on aten/src/ATen/core/interned_strings.h:
```diff
@@ -0,0 +1,279 @@
+
+namespace torch { namespace jit {
+
+#if !AT_MOBILE
```
This deserves a comment, I think.

bwasti commented:
@pytorchbot retest this please

bwasti commented:
@pytorchbot retest this please

bwasti committed:
Move interned_strings and get build working (#12039)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/12039

Refactoring out this diff D9819906

Reviewed By: ezyang

Differential Revision: D10024844

fbshipit-source-id: b4b32160f679a7c51a8089625b824256da75c6d2
